### DISCUSSION TITLE
Updating values into BACnet objects

### DISCUSSION THREAD

**Discussion by Aleksoski1983** [ChristianTremblay/BAC0#502] [Q&A] [answered]
Hello Guys,
I'm trying to create a simple code in Python using BAC0 library to simulate a BACNet device. There is some issue about updating values assigned in object definition. So, the value (new_value) might update the object locally in the script, but this change not reflecting in the BACnet network. There is no assignment of value in defined object. I attached picture where this is indicated. 

If you have any suggestions,I will appreciate that.

![BACNet server script](https://github.com/user-attachments/assets/c0fa15ae-2b72-4c89-8e38-24a39c7741b7)

**Comment by ChristianTremblay** [comment, marked as answer]
you can't reference your object like this. The add_object_to_application method will put everything where bacpypes needs it. The thing you get back from this is an object that contains all the object you want to build....

```python
device1['ZN-T'].presentValue = new_value
```
should work
And you don't need to clear_objects in your loop.

Also, have a look to the tasks (RecurringTask). This way you can create different functions to update the value of your objects and the execution will be automatically dealt by the task manager

```python
    # extract from code...
    _new_objects.add_objects_to_application(weather_device)
    app = WeatherApp(weather_device)
    task_device = RecurringTask(
            update, delay=int(os.environ.get("OWM_REFRESHRATE"))
        )
    task_device.start()
    while True:
        await asyncio.sleep(0.1)
```

I'm not sure, but my instinct would tell me not to wait 5 seconds in the main loop as it could create delays in the responses BAC0 will give to other BACnet device talking to it. I know it is async code, but I'm pretty sure `await asyncio.sleep` will block for 5 seconds before continuing... (I may be wrong). Using a task will eliminate risks.

**Comment by Aleksoski1983** [comment]
Thank you @ChristianTremblay for you fast response. First line above works and updates a value. Thanks for your help.
